presto-orc/src/main/java/com/facebook/presto/orc/writer/SliceDictionaryColumnWriter.java

```diff
@@ -14,603 +14,300 @@
+            LongOutputStream dataStream)
     {
-        checkState(closed);
-
-        if (directEncoded) {
-            return directColumnWriter.getIndexStreams();
-        }
-
-        ImmutableList.Builder<RowGroupIndex> rowGroupIndexes = ImmutableList.builder();
-
-        List<LongStreamCheckpoint> dataCheckpoints = dataStream.getCheckpoints();
-        Optional<List<BooleanStreamCheckpoint>> presentCheckpoints = presentStream.getCheckpoints();
-        for (int i = 0; i < rowGroups.size(); i++) {
-            int groupId = i;
-            ColumnStatistics columnStatistics = rowGroups.get(groupId).getColumnStatistics();
-            LongStreamCheckpoint dataCheckpoint = dataCheckpoints.get(groupId);
-            Optional<BooleanStreamCheckpoint> presentCheckpoint = presentCheckpoints.map(checkpoints -> checkpoints.get(groupId));
-            List<Integer> positions = createSliceColumnPositionList(compressionParameters.getKind() != NONE, dataCheckpoint, presentCheckpoint);
-            rowGroupIndexes.add(new RowGroupIndex(positions, columnStatistics));
+        checkState(optionalSortedIndex.isPresent() == sortDictionaryKeys, "SortedIndex and sortDictionaryKeys(%s) are inconsistent", sortDictionaryKeys);
+        for (int position = 0; position < rowGroupValueCount; position++) {
+            presentStream.writeBoolean(rowGroupIndexes[position] != NULL_INDEX);
         }
 
-        Slice slice = compressedMetadataWriter.writeRowIndexes(rowGroupIndexes.build());
-        Stream stream = new Stream(column, StreamKind.ROW_INDEX, slice.length(), false);
-        return ImmutableList.of(new StreamDataOutput(slice, stream));
-    }
-
-    private static List<Integer> createSliceColumnPositionList(
-            boolean compressed,
-            LongStreamCheckpoint dataCheckpoint,
-            Optional<BooleanStreamCheckpoint> presentCheckpoint)
-    {
-        ImmutableList.Builder<Integer> positionList = ImmutableList.builder();
-        presentCheckpoint.ifPresent(booleanStreamCheckpoint -> positionList.addAll(booleanStreamCheckpoint.toPositionList(compressed)));
-        positionList.addAll(dataCheckpoint.toPositionList(compressed));
-        return positionList.build();
-    }
-
-    @Override
-    public List<StreamDataOutput> getDataStreams()
-    {
-        checkState(closed);
-
-        if (directEncoded) {
-            return directColumnWriter.getDataStreams();
+        if (sortDictionaryKeys) {
+            int[] sortedIndexes = optionalSortedIndex.get();
+            for (int position = 0; position < rowGroupValueCount; position++) {
+                int originalDictionaryIndex = rowGroupIndexes[position];
+                if (originalDictionaryIndex != NULL_INDEX) {
+                    int sortedIndex = sortedIndexes[originalDictionaryIndex];
+                    if (sortedIndex < 0) {
+                        throw new IllegalArgumentException(String.format("Invalid index %s at position %s", sortedIndex, position));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: sortedIndex (declared at line 246), position (declared at line 243). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 248. Consider formatting this data with &#39;%d&#39; instead.